simd/unit_tests/include/TestSIMD_ShiftOps.hpp

```diff
@@ -245,13 +256,23 @@ KOKKOS_INLINE_FUNCTION void device_check_shift_ops() {
 
       device_check_shift_op_all_loaders<Abi>(shift_right(), test_vals, shift_by,
                                              num_cases);
+      device_check_shift_op_all_loaders<Abi>(shift_right_eq(), test_vals,
+                                             shift_by, num_cases);
       device_check_shift_op_all_loaders<Abi>(shift_left(), test_vals, shift_by,
                                              num_cases);
+      device_check_shift_op_all_loaders<Abi>(shift_left_eq(), test_vals,
+                                             shift_by, num_cases);
 
       if constexpr (std::is_signed_v<DataType>) {
         for (std::size_t i = 0; i < width; ++i) test_vals[i] *= -1;
         device_check_shift_op_all_loaders<Abi>(shift_right(), test_vals,
                                                shift_by, num_cases);
+        device_check_shift_op_all_loaders<Abi>(shift_right_eq(), test_vals,
+                                               shift_by, num_cases);
+        device_check_shift_op_all_loaders<Abi>(shift_left(), test_vals,
+                                               shift_by, num_cases);
+        device_check_shift_op_all_loaders<Abi>(shift_left_eq(), test_vals,
+                                               shift_by, num_cases);
```

Comment:
Left shifting negative values is undefined behavior (as the CI detects as well).
```suggestion
```

Reply:
Right shifting negative values is also UB: [[expr.shift]](https://eel.is/c++draft/expr.shift#1)

Reply:
cppreference says in https://en.cppreference.com/w/cpp/language/operator_arithmetic:
> For negative a, the value of a >> b is implementation-defined (in most implementations, this performs arithmetic right shift, so that the result remains negative). 

Doesn't [[expr.shift]](https://eel.is/c++draft/expr.shift#1) only talk about the right operand?

Reply:
You are correct; it only talks about the right operand.

FYI: somewhere between C++17 and the current WD, right shifting negative numbers became well-defined (It shouldn't matter for this PR though).

Reply:
Removed tests for left shifting signed values.
For right shifting, right operands are set to be 0 or positive values in this test, so shouldn't be a problem.